In [27]:
from tensorflow.keras.applications import ResNet50

In [28]:
import tensorflow as tf
from tensorflow.keras.datasets import cifar10
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Flatten, Dense, Dropout, BatchNormalization

In [29]:
#załadaowanie zbioru cifar-10
(X_train, y_train), (X_test, y_test) = cifar10.load_data()

In [30]:
#normalizacja danych na zakres [0,1]
X_train = X_train.astype('float32') / 255.0 
X_test = X_test.astype('float32') / 255.0

In [31]:
#załadowanie modelu z określonum kształtem wejściowym
model = ResNet50(weights='imagenet', include_top=False, input_shape=(32, 32, 3)) 
model.trainable = False #warstwa bazowa nieedytowalna


In [ ]:
model = Sequential([
    model,
    Flatten(), #splaszczenie danych wyjsciowych z warstwy resnet50
    Dense(512, activation='relu'), #512 neuronow aktywacja relu
    BatchNormalization(), #normalizacja batch
    Dropout(0.5), #regularyzacja dropoutem na 0.5
    Dense(256, activation='relu'), #256neuronów relu
    BatchNormalization(),
    Dropout(0.5),
    Dense(10, activation='softmax') #database ma 10 klas
])

In [ ]:
#kompilacja, optymalizator adam
model.compile(optimizer=tf.keras.optimizers.Adam(learning_rate=0.0001), loss='sparse_categorical_crossentropy', metrics=['accuracy'])

In [34]:
model.fit(X_train, y_train, epochs=10) #trenowanie przez 10 epok
# Ewaluacja modelu
loss, accuracy = model.evaluate(X_test, y_test) #ewaluacja na danych testowych
print(f'Test Accuracy: {accuracy}')

# Zapisanie najlepszego modelu do pliku
model.save('ffinal_model.h5')


Epoch 1/10
1563/1563 [==============================] - 28s 17ms/step - loss: 2.5603 - accuracy: 0.2421
Epoch 2/10
1563/1563 [==============================] - 27s 17ms/step - loss: 2.1491 - accuracy: 0.3046
Epoch 3/10
1563/1563 [==============================] - 26s 17ms/step - loss: 1.9630 - accuracy: 0.3354
Epoch 4/10
1563/1563 [==============================] - 26s 17ms/step - loss: 1.8466 - accuracy: 0.3569
Epoch 5/10
1563/1563 [==============================] - 26s 17ms/step - loss: 1.7649 - accuracy: 0.3801
Epoch 6/10
1563/1563 [==============================] - 26s 16ms/step - loss: 1.7100 - accuracy: 0.3942
Epoch 7/10
1563/1563 [==============================] - 23s 15ms/step - loss: 1.6743 - accuracy: 0.4042
Epoch 8/10
1563/1563 [==============================] - 22s 14ms/step - loss: 1.6437 - accuracy: 0.4173
Epoch 9/10
1563/1563 [==============================] - 22s 14ms/step - loss: 1.6205 - accuracy: 0.4210
Epoch 10/10
313/313 [==============================] - 4s 12ms/s

wnioski:
dokładność 22% jest bardzo niska i może to być wynikiem wielu rzeczy między innymi:

-brak optymalizacji danych względem modelu, resnet50 moze byc zbyt skomplikowany na zbior danych cifar-10

-zbyt mała ilośc epok

-źle dobrany wstępnie wytrenowany model względem wybranych danych